src/storage/heap_file.c

```diff
@@ -24221,6 +24214,20 @@ heap_get_visible_version_internal (THREAD_ENTRY * thread_p, HEAP_GET_CONTEXT * c
       context->class_oid_p = &class_oid_local;
     }
 
+  if (context->scan_cache && context->scan_cache->area == NULL && context->ispeeking == COPY
```

Comment:
I'd like to propose to introduce a function to allocate `scancache->area` to avoid redundancies.